[TRACING-3763] Fix the documentation about OTEL for multicluster and certs (#69690)
openshift/openshift-docs: modules/otel-config-multicluster.adoc

[id="gathering-observability-data-from-different-clusters_{context}"]
= Gathering the observability data from different clusters with the OpenTelemetry Collector

> For a multicluster configuration, you can create one OpenTelemetry Collector instance in each one of the remote clusters and then forward all the telemetry data to one OpenTelemetry Collector instance.

Reviewer: skrthomas, I'd recommend preferring the core OpenShift pages, which use `multicluster`, rather than the aligned product pages: https://docs.openshift.com/search.html?query=multicluster. Also, IBM Style returns only `multicluster` results and none of `multi-cluster` results.

> For a multicluster configuration, you can create one OpenTelemetry Collector instance in each one of the remote clusters and then forward all the telemetry data to one OpenTelemetry Collector instance.

Reviewer 2: Fair enough, the repo goes both ways it seems, and you're right that IBMSG doesn't hyphenate. I'm fine with leaving it as it is.

.Prerequisites

* The {OTELOperator} is installed.
* The {TempoOperator} is installed.

> * A TempoStack instance is deployed on the cluster.

Reviewer: @skrthomas, yes, this too was my first suggestion to @iblancasa :sweat_smile: until I got to the certificate-mounting steps. My concern with adding these as prerequisites is that many readers will stop reading at the list of prerequisites, then waste their time looking up this information elsewhere, and when they finally get to the procedure, they'll be upset that we give them the instructions after they've spent time looking for this information elsewhere.

> * A TempoStack instance is deployed on the cluster.

Reviewer 2: I can see your concern here. I guess, these things are still prerequisite, so would a customer get to step 1 and think, none of these things were listed in the prerequisites but I'm being asked to do them now? Maybe you can add to the end of my suggested prerequisite line "Steps for creating these certificates are outlined in step 1." or something like that, to offset customers charting off to search on their own.

> * A TempoStack instance is deployed on the cluster.

Reviewer: @skrthomas, WDYT?
```suggestion
* A TempoStack instance is deployed on the cluster.
* The following mounted certificates: Issuer, self-signed certificate, CA issuer, client and server certificates. To create any of these certificates, see step 1.
```

> * A TempoStack instance is deployed on the cluster.

Reviewer 2: This looks good! Thank you @max-cx

.Procedure

> . You must have the following certificates mounted as secrets in the OpenTelemetry Collector instance. If you don't have them, create them.

Reviewer: @iblancasa,  :smiley: what's your rough estimated probability of all of these certificates already mounted in the reader's OTEL instance? If it's more likely that the reader has to mount at least some of these certificates, then my suggestion is to keep it as a step with possible exceptions. Here's a suggestion with revised wording of the step:
```suggestion
. Mount the following certificates in the OpenTelemetry Collector instance, skipping already mounted certificates.
```

> . You must have the following certificates mounted as secrets in the OpenTelemetry Collector instance. If you don't have them, create them.

Reviewer: @iblancasa,  :smiley: what's your rough estimated probability of all of these certificates already mounted in the reader's OTEL instance? If it's more likely that the reader has to mount at least some of these certificates, then my suggestion is to keep it as a step with possible exceptions. Here's a suggestion with revised wording of the step:
```suggestion
. Mount the following certificates in the OpenTelemetry Collector instance, skipping already mounted certificates.
```

> . You must have the following certificates mounted as secrets in the OpenTelemetry Collector instance. If you don't have them, create them.

Author: The user needs to mount them. They will not be mounted by default. So, I'll add your suggestion.

    exporters:
      otlphttp:
        endpoint: https://observability-cluster.com:443 # <1>
        tls:
          insecure: false
          cert_file: /certs/server.crt
          key_file: /certs/server.key
          ca_file: /certs/ca.crt
    service:
      pipelines:
        traces:
          receivers: [jaeger, opencensus, otlp, zipkin]
          processors: [memory_limiter, k8sattributes, resourcedetection, batch]
          exporters: [otlp]
  volumes:
    - name: otel-certs
      secret: 
        name: otel-certs
  volumeMounts:
    - name: otel-certs
      mountPath: /certs
----

> <1> The Collector exporter is configured to export OTLP HTTP and points to the OpenTelemetry Collector from the central cluster. Note you need to provide the certificate as a secret and mount it in your OpenTelemetry Collector instance. You can create the certificate by hand or using the {cert-manager-operator}.

Reviewer: @iblancasa, what kind of certificate is this about?

Author: This needs:
- A TLS cert (`cert_file`)
- A TLS key (`key_file`)
- A CA cert (`ca_file`)

Reviewer: Is it possible that these certs will be already created by some users for other purposes?
Or do these certs have to be created for every installed OTEL instance?

Author: They could have create them before.

Reviewer: @iblancasa, WDYT if I add this to the Prerequisites and if I delete the second sentence on line 128?
 
```
* You have the following certificates mounted as secrets in your {OTEL} instance:
+
[TIP]
====
You can create these certificates manually or by using the {cert-manager-operator}.
====
+
** TLS certificate, for example as a `cert_file`.
** TLS key, for example as a `key_file`.
** CA certificate, for example as a `ca_file`.
```

Author: LGTM
